File: src/main/kotlin/basic_control_structures/exercise_12.kt
package basic_control_structures

// Write a program that reads the price of a product, calculates and shows its PROMOTIONAL PRICE, with a 5% discount.
fun promotionalPrice() {
    print("price: ")
    val price = readln().toDouble()
    val _price = String.format("%.2f", price)

    print("desconto(%): ")
    val discountPercentage = readln().toDouble()

    val discount = discountPercentage * price / 100
    val _discount = String.format("%.2f", discount)

    val discountedPrice = price - discount
    val _discountedPrice = String.format("%.2f", discountedPrice)

    println("Price: R$$_price, discount: R$$_discount, discounted price: R$$_discountedPrice.")
}

File: Dart/a_basic_control_structure/exercise_12.dart
// Write a program that reads the price of a product, calculates and shows its PROMOTIONAL PRICE, with a 5% discount.

import 'dart:io';

void main() {
  String answer = 'yes';

  while (answer == 'yes') {
    try {
      stdout.write('product price: ');
      double productPrice = double.parse(stdin.readLineSync()!);

      stdout.write('discount(%): ');
      double percentage = double.parse(stdin.readLineSync()!);

      promotionalPrice(productPrice, percentage);

      stdout.write('do you want to try again? ');
      answer = stdin.readLineSync()!.toLowerCase();
    } catch (exception) {
      print('Error: Please enter a valid integer');
    }
  }
}

double promotionalPrice(double productPrice, double percentage) {
  double convertedValue = percentage / 100;
  double discountValue = productPrice * convertedValue;
  double finalPrice = productPrice - discountValue;

  print(
      'Product price: US\$${productPrice.toStringAsFixed(2)}, discount: US\$${discountValue.toStringAsFixed(2)}, final price: US\$${finalPrice.toStringAsFixed(2)}');

  return finalPrice;
}
